pkg/dao/external_resource.go

```diff
@@ -46,3 +80,17 @@ func UrlToRepomdUrl(urlIn string) (string, error) {
 	u.Path = path.Join(u.Path, "/repodata/repomd.xml")
 	return u.String(), nil
 }
+
+func validateGpgKeyString(bodyString string) error {
+	httpError := echo.NewHTTPError(http.StatusNotAcceptable, "Received response is not a valid gpg-key")
```

Comment:
Having echo in the dao layer feels illegal. Maybe this can be done in such a way that echo is not imported here?

Reply:
Looking in `dao/repository_config.go`, we use this [error struct](https://github.com/content-services/content-sources-backend/blob/main/pkg/dao/error.go) along with this [DBErrorToApi](https://github.com/content-services/content-sources-backend/blob/main/pkg/dao/repository_config.go#L26) function. Then in the handler we use this [httpCodeForError](https://github.com/content-services/content-sources-backend/blob/main/pkg/handler/repositories.go#L136) function to map that to an http error. I think you could leverage the same functions here.

Reply:
That seems like some workaround. I have used a different package and caught the error higher up in the tree to avoid this instead.